limitador/src/storage/redis/scripts.rs

```diff
@@ -14,7 +14,7 @@
 pub const SCRIPT_UPDATE_COUNTER: &str = "
     local c = redis.call('incrby', KEYS[1], ARGV[2])
     if c == tonumber(ARGV[2]) then
-      redis.call('expire', KEYS[1], ARGV[1], 'NX')
+      redis.call('expire', KEYS[1], ARGV[1])
```

Comment:
According to redis doc https://redis.io/commands/expire/
```
History

    Starting with Redis version 7.0.0: Added options: NX, XX, GT and LT.

```

So, removing the `NX` option to keep compatibility with redis 5 and redi 6.

Actually, I think it is not needed, as the key has just been created, so expiration has not been set. This happens within the context of the script, so no race condition issues can hit here.

Reply:
Strictly speaking we indeed don't need it...